ui/packages/mui-extends/src/Checkbox/index.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Chaos Mesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { FormControl, FormControlLabel, FormHelperText, FormLabel } from '@mui/material'
+import { default as MuiCheckbox, CheckboxProps as MuiCheckboxProps } from '@mui/material/Checkbox'
+
+export type CheckboxProps = MuiCheckboxProps & {
+  /**
+   * The name of ths input field when using it in a form.
+   */
+  name?: string
+  /**
+   * The label of the checkbox, would be shown on the top of the checkbox.
+   */
+  label: string
+  /**
+   * The helper text of the checkbox, would be shown on the right of the checkbox in the same line.
+   *
+   * If not provided, it would use the `label` as the helper text.
+   */
+  helperText?: string
+  /**
+   * Validation error message, would be shown on the bottom of the checkbox.
+   */
+  errorMessage?: string
+}
+
+export default (props: CheckboxProps) => {
+  return (
+    <div>
+      <FormControl error={props.errorMessage !== ''} required={true}>
+        <FormLabel component="legend">{props.label}</FormLabel>
+        <FormControlLabel
+          control={
+            <MuiCheckbox
+              checked={props.checked}
+              onChange={props.onChange}
+              disabled={props.disabled}
+              inputProps={{ name: props.name }}
+            ></MuiCheckbox>
+          }
+          label={props.helperText || props.label}
+        ></FormControlLabel>
+
+        <FormHelperText>{props.errorMessage}</FormHelperText>
```

Comment:
I think you wanna render an optional error message when a checkbox has an error? 🤔 Then I think the following would be a better way:

```jsx
<FormHelperText error={props.error}>
  {error && props.errorMessage ? props.errorMessage : props.helperText}
</FormHelperText>
```

Reply:
That could work with most other components like TextField.  It's also related to the prototype, if the text right beside of checkbox means `helpText`, I think maybe showing errorMessage in other places is better.

Another thing is we do not have both `error` and `errorMessage`, we have only `errorMessage` now. Maybe declaring both of them is better?

Reply:
> That could work with most other components like TextField. It's also related to the prototype, if the text right beside of checkbox means `helpText`, I think maybe showing errorMessage in other places is better.
> 
> Another thing is we do not have both `error` and `errorMessage`, we have only `errorMessage` now. Maybe declaring both of them is better?

Follow https://mui.com/components/checkboxes/#label, it means a `label`, not a `helperText`. I can provide an example of how the mui represents an error when it has occurred:

https://github.com/chaos-mesh/chaos-mesh/blob/821d65d78e45eb340cec9816154b0b4f86ba1c48/ui/app/src/components/NewExperimentNext/Step2.tsx#L148

Usually, the `helperText` will perform two responsibilities, one for rendering a helper text and another is rendering an error message when an error occurred.

And the `error` will tell a form component if there has an error. You can find the `error` props in https://mui.com/api/text-field/.

Reply:
Got that,  I would update later.

Reply:
Updated! PTAL! :D